pkg/scaffold/resource/resource_test.go

```diff
@@ -26,123 +28,158 @@ import (
 var _ = Describe("Resource", func() {
 	Describe("scaffolding an API", func() {
 		It("should succeed if the Resource is valid", func() {
-			instance := &Resource{Group: "crew", Version: "v1", Kind: "FirstMate"}
-			Expect(instance.Validate()).To(Succeed())
+			options := &Options{Group: "crew", Version: "v1", Kind: "FirstMate"}
+			Expect(options.Validate()).To(Succeed())
+
+			resource := options.NewResource("test", "test.io")
+			Expect(resource.Namespaced).To(Equal(options.Namespaced))
+			Expect(resource.Group).To(Equal(options.Group))
+			Expect(resource.GroupPackageName).To(Equal("crew"))
+			Expect(resource.Version).To(Equal(options.Version))
+			Expect(resource.Kind).To(Equal(options.Kind))
+			Expect(resource.Plural).To(Equal("firstmates"))
+			Expect(resource.ImportAlias).To(Equal("crewv1"))
+			// Note: The following two tests will work because, despite not
+			// finding the file, "crew" is not of of the core groups.
+			Expect(resource.Package).To(Equal(path.Join("test", "api", "v1")))
+			Expect(resource.Domain).To(Equal("crew.test.io"))
+			Expect(resource.CreateExampleReconcileBody).To(Equal(false))
+
+			resource = options.NewV1Resource("test", "test.io")
+			Expect(resource.Namespaced).To(Equal(options.Namespaced))
+			Expect(resource.Group).To(Equal(options.Group))
+			Expect(resource.GroupPackageName).To(Equal("crew"))
+			Expect(resource.Version).To(Equal(options.Version))
+			Expect(resource.Kind).To(Equal(options.Kind))
+			Expect(resource.Plural).To(Equal("firstmates"))
+			Expect(resource.ImportAlias).To(Equal("crewv1"))
+			// Note: The following two tests will work because, despite not
+			// finding the file, "crew" is not of of the core groups.
+			Expect(resource.Package).To(Equal(path.Join("test", "pkg", "apis", "crew", "v1")))
+			Expect(resource.Domain).To(Equal("crew.test.io"))
```

Comment:
IHMO GroupDomain make more intuitive its understanding but it is a nit.

Reply:
It is both the Resource and the Group domain, and as we are already in the `Resource` struct i left it as `Domain`. Let's see if others agree with you and I'll change it if so.